gobblin-cluster/src/main/java/org/apache/gobblin/cluster/GobblinHelixJobScheduler.java

```diff
@@ -324,10 +332,18 @@ public void handleUpdateJobConfigArrival(UpdateJobConfigArrivalEvent updateJobAr
   }
 
   @Subscribe
-  public void handleDeleteJobConfigArrival(DeleteJobConfigArrivalEvent deleteJobArrival) {
+  public void handleDeleteJobConfigArrival(DeleteJobConfigArrivalEvent deleteJobArrival) throws InterruptedException {
     LOGGER.info("Received delete for job configuration of job " + deleteJobArrival.getJobName());
     try {
       unscheduleJob(deleteJobArrival.getJobName());
+      Properties jobConfig = deleteJobArrival.getJobConfig();
```

Comment:
This should go to a private method cancelJobIfRequired() coupled with a public util method cancelHelixJob

Reply:
Added a method.